src/test/java/com/puppycrawl/tools/checkstyle/grammar/java21/Java21AstRegressionTest.java

```diff
@@ -46,6 +46,15 @@ public void testBasicStringTemplate() throws Exception {
                         "InputStringTemplateBasic.java"));
     }
 
+    /**
+     * The purpose of this test is to exercise the
+     * {@link com.puppycrawl.tools.checkstyle.grammar.CompositeLexerContextCache}:
+     * getting some curly brace contexts on the stack, making sure that each
+     * element in the stack has some depth. This ensures that we can handle
+     * many nested curly braces within an embedded template expression.
+     *
+     * @throws Exception upon failure
+     */
     @Test
     public void testStringTemplateNested() throws Exception {
```

Comment:
I've extended this test case to get our nesting level of curly braces even deeper within one "context", and left a comment here about this test case.